### DISCUSSION TITLE
How to prevent logging "Icon url is missing." error?

### DISCUSSION THREAD

**Discussion by fifzteen** [visgl/deck.gl#5661] [Q&A] [answered]
Hi.
I use IconLayer with setting Promise to `iconAtals` and `iconMapping`.
While resolved that promises, "Icon url is missing." error is logging to console.

```
const atlas = getAtlas() // returns Promise
const mapping = getMapping() // returns Promise

new IconLayer({
  id: 'icon',
  iconAtlas: atlas,
  iconMapping: mapping,
  getIcon: (d) => d.icon
})
```

How to prevent logging "Icon url is missing." error?

**Comment by padawannn** [comment, marked as answer]
Hi @fifzteen,
I've done a quick test and I cannot see the logging errors. Could you send us an example?

**Reply by fifzteen** [reply]
@padawannn
Thank you for your checking. 
I make an example, but I cannot see that error, too.
https://codepen.io/fifzteen/pen/abpKzwG

Maybe there's another reason for the system I'm really developing.
I'll do some more research on my own.
